inspirehep/modules/workflows/tasks/manual_merging.py

```diff
@@ -0,0 +1,157 @@
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Tasks related to manual merge generic record."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+
+from invenio_db import db
+
+from inspire_dojson.utils import get_record_ref
+from inspire_json_merger.api import merge
+from inspirehep.modules.workflows.utils import insert_wf_record_source
+from inspirehep.modules.workflows.utils import with_debug_logging
+from inspirehep.utils.record_getter import get_db_record
+
+
+@with_debug_logging
+def merge_records(obj, eng):
+    """Perform a manual merge.
+
+    Merges two records stored in the workflow object as the content of the
+    ``head`` and ``update`` keys, and stores the result in ``obj.data``.
+    Also stores the eventual conflicts in ``obj.extra_data['conflicts']``.
+
+    Because this is a manual merge we assume that these two records have no
+    common ancestor, so ``root`` is the empty dictionary.
+
+    Args:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        None
+
+    """
+    head, update = obj.extra_data['head'], obj.extra_data['update']
+    head_source = obj.extra_data['head_source']
+
+    merged, conflicts = merge(
+        root={},
+        head=head,
+        update=update,
+        head_source=head_source,
+    )
+
+    obj.data = merged
+    obj.extra_data['conflicts'] = [json.loads(el.to_json()) for el in conflicts]
+    obj.save()
+
+
+@with_debug_logging
+def halt_for_approval(obj, eng):
+    """Stop the Workflow engine.
+
+    Pauses the workflow using the 
+
+    This pause the workflow using the custom action `merge_approval`, that is
+    used every time there are merging conflicts to solve.
+
+    Args:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        None
+
+    """
+    eng.halt(
+        action='merge_approval',
+        msg='Manual Merge halted for curator approval.'
+    )
+
+
+@with_debug_logging
+def edit_metadata_and_store(obj, eng):
+    """Store the records involved in the manual merging process.
+
+    Delete the ``update``, redirecting it to the merged record. Add a
+    reference in the merged record to the deleted one, and store them in the db.
+
+    Args:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        None
+
+    """
+    head = get_db_record('lit', obj.extra_data['head_control_number'])
+    update = get_db_record('lit', obj.extra_data['update_control_number'])
+
+    head.clear()
+    head.update(obj.data)    # head's content will be replaced by merged
+    update.merge(head)       # update's uuid will point to head's uuid
+    update.delete()          # mark update record as deleted
+
+    # add schema contents to refer deleted record to the merged one
+    update['new_record'] = get_record_ref(
+        head['control_number'],
+        endpoint='record'
+    )
```

Comment:
This is not right, as it would generate a JSON reference like
```JSON
{"$ref": "http://localhost:5000/api/record/4328"}
```
which isn't correct, because we don't redirect routes like `/api/record/<int:control_number>`, just `/record/<int:control_number>`.

Reply:
As other parts of the code assume that we're merging Literature records (most notably, we don't have configuration otherwise in `inspire-json-merger`) it's ok to hardcode it here too.